File: Design Patterns/Singleton Pattern/Java/com.faculdadevincit.desingpatterns.singletonpattern/src/ContadorSingleton.java
public class ContadorSingleton {
    private int valor = 0;
    private static ContadorSingleton instance;

    public static ContadorSingleton getInstance() {
        if (instance == null) {
            instance = new ContadorSingleton();
        }
        return instance;
    }

    public ContadorSingleton() {
        super();
    }

    public int getValor() {
        return valor;
    }

    public void incrementar() {
        valor++;
    }
}

File: Design Patterns/Singleton Pattern/Python/contador_singleton.py
class ContadorSingleton:
    _valor = 0
    _instance = None

    @property
    def valor(self):
        return self._valor

    def incrementar(self):
        self._valor += 1

    def __new__(self):
        if self._instance is None:
            self._instance = super(ContadorSingleton, self).__new__(self)
        return self._instance

        
